corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -85,15 +95,26 @@ def _migrate_form_and_associated_models(self, couch_form):
         _migrate_form_attachments(sql_form, couch_form)
         _migrate_form_operations(sql_form, couch_form)
 
-        diffs = json_diff(couch_form.to_json(), sql_form.to_json(), track_list_indices=False)
+        couch_form_json = couch_form.to_json()
+        sql_form_json = sql_form.to_json()
+        diffs = json_diff(couch_form_json, sql_form_json, track_list_indices=False)
         self.diff_db.add_diffs(
             couch_form.doc_type, couch_form.form_id,
-            filter_form_diffs(couch_form.doc_type, diffs)
+            filter_form_diffs(couch_form_json, sql_form_json, diffs)
         )
 
         case_stock_result = None
         if sql_form.initial_processing_complete:
             case_stock_result = _get_case_and_ledger_updates(self.domain, sql_form)
+            if len(case_stock_result.case_models):
+                touch_updates = [
+                    update for update in get_case_updates(couch_form)
+                    if len(update.actions) == 1 and isinstance(update.actions[0], CaseNoopAction)
+                ]
+                if len(touch_updates):
```

Comment:
nitpick: [pep8 recommends](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)* `if touch_updates:`, although it may be even more efficient to use a generator expression with `any()` since that will return True on the first item encountered.

\* search for `if len(seq):`